### PR TITLE
Allow app.yaml to be generated by gcloud when deploying

### LINKED ISSUE #131: Allow app.yaml to be automatically generated when deploying

### PR DESCRIPTION
**Pull request by meltsufin**
Fixes #131.

This is pending fix in appengine-plugins-core: https://github.com/GoogleCloudPlatform/appengine-plugins-core/issues/317

### CODE REVIEW

File: src/main/java/com/google/cloud/tools/maven/DeployMojo.java [hunk 1]
```diff
@@ -93,9 +93,11 @@ public void execute() throws MojoExecutionException, MojoFailureException {
...
     super.execute();
 
-    if (deployables.size() == 0) {
-      deployables.add(
-          new File(getStagingDirectory() + "/app.yaml"));
+    File appYamlFile = new File(getStagingDirectory() + "/app.yaml");
+    if (deployables.size() == 0 && appYamlFile.exists()) {
+      deployables.add(appYamlFile);
+    } else {
+      deployables.add(stagingDirectory);
```
Comment (joaoandremartins) [inline src/main/java/com/google/cloud/tools/maven/DeployMojo.java]: Why don't we stage an app.yaml file instead, if none exists?

Comment (meltsufin) [inline src/main/java/com/google/cloud/tools/maven/DeployMojo.java]: I don't think we want to maintain yet another app.yaml generation template.
Here we're just letting the Cloud SDK generate the app.yaml based on its fingerprinting logic.

Comment (joaoandremartins) [inline src/main/java/com/google/cloud/tools/maven/DeployMojo.java]: I don't feel very strongly either way, so sure

Comment (chanseokoh) [inline src/main/java/com/google/cloud/tools/maven/DeployMojo.java]: Question: you use `getStagingDirectory()` above, but here, it's just `stagingDirectory`. This makes me wonder if there is a difference.

Comment (meltsufin) [inline src/main/java/com/google/cloud/tools/maven/DeployMojo.java]: No difference, but good catch, the code should use consistent access.

**Pull request merged**